### PR TITLE
Implemented login flow(#2h198uc)

### PR DESCRIPTION
**Pull request by disha1202**

### CODE REVIEW

File: src/services/UserService.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    method: "post",
+    data: {
+      'USERNAME': username, 
+      'PASSWORD': password
+    }
+  });
+}
+
+const getProfile = async (): Promise <any>  => {
+    return api({
```
Comment (ymaheshwari1) [inline src/services/UserService.ts:15]: Make indentation consistent.

File: src/store/modules/user/UserState.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default interface ShopState {
```
Comment (ymaheshwari1) [inline src/store/modules/user/UserState.ts:1]:
```suggestion
export default interface UserState {
```

File: src/store/modules/user/index.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    instanceUrl: '',
+    current: null,
+    currentFacility: {},
+  },
+  getters,
+  actions,
+  mutations,
+}
+
+export default shopModule;
```
Comment (ymaheshwari1) [inline src/store/modules/user/index.ts:21]:
```suggestion
export default userModule;
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts (hunk 1 above)
Comment (disha1202) [reply src/services/UserService.ts]: Improved indentation

File: src/store/modules/user/UserState.ts (hunk 2 above)
Comment (disha1202) [reply src/store/modules/user/UserState.ts]: Made the requested change

File: src/store/modules/user/index.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import actions from './actions'
+import getters from './getters'
+import mutations from './mutations'
+import { Module } from 'vuex'
+import UserState from './UserState'
+import RootState from '@/store/RootState'
+
+const shopModule: Module<UserState, RootState> = {
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/store/modules/user/index.ts]: Made the requested change

File: src/store/modules/user/index.ts (hunk 3 above)
Comment (disha1202) [reply src/store/modules/user/index.ts]: Made the requested change

File: src/views/Login.vue [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        <form class="login-container" @keyup.enter="login(form)" @submit.prevent="login(form)">
+          <Logo />
+
+          <ion-item lines="full" v-if="!baseURL">
+            <ion-label position="fixed">{{ $t("OMS") }}</ion-label>
+            <ion-input name="instanceUrl" v-model="instanceUrl" id="instanceUrl"  type="text" required />
+          </ion-item>
+          <ion-item lines="full">
+            <ion-label position="fixed">{{ $t("Username") }}</ion-label>
+            <ion-input name="username" v-model="username" id="username"  type="text" required></ion-input>
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/views/Login.vue]: Used self closing tag.

File: src/store/modules/user/actions.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  async login ({ commit, dispatch }, { username, password }) {
+    try {
+      const resp = await UserService.login(username, password)
+      if (resp.status === 200 && resp.data) {
+        if (resp.data.token) {
+          commit(types.USER_TOKEN_CHANGED, { newToken: resp.data.token })
+          await dispatch('getProfile')
+          return resp.data;
+        } else if (hasError(resp)) {
+          showToast(translate('Sorry, your username or password is incorrect. Please try again.'));
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:19]: Add it to en.json.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 6 above)
Comment (disha1202) [reply src/store/modules/user/actions.ts:19]: Added to en.json

**Review by ymaheshwari1 — APPROVED** [review]

File: src/router/index.ts [hunk 7]
```diff
@@ -1,13 +1,39 @@
...
+  }
+};
+
 const routes: Array<RouteRecordRaw> = [
   {
     path: '/install',
     name: 'Install',
-    component: Install
+    component: Install,
+    beforeEnter: authGuard,
```
Comment (adityasharma7) [inline src/router/index.ts:30]: We should not add any guard to install page

File: src/views/Login.vue [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    ...mapGetters({
+      currentInstanceUrl: 'user/getInstanceUrl'
+    })
+  },
+  mounted() {
+    this.instanceUrl = this.currentInstanceUrl;
+  },
+  methods: {
+    login: function () {
+      if(!this.baseURL) this.store.dispatch("user/setUserInstanceUrl", this.instanceUrl.trim())
```
Comment (adityasharma7) [inline src/views/Login.vue:74]: We will use the instanceURL from shop config VUE_APP_SHOPIFY_SHOP_CONFIG

File: src/views/Settings.vue [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    IonItem, 
+    IonLabel, 
+    IonList,
+    IonPage,  
+    IonTitle,
+    IonToolbar
+  },
+  data(){
+    return {
+      baseURL: process.env.VUE_APP_BASE_URL
```
Comment (adityasharma7) [inline src/views/Settings.vue:51]: We will not store VUE_APP_BASE_URL in configuration in any case. We could remove this

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/router/index.ts [hunk 10]
```diff
@@ -1,23 +1,55 @@
...
     path: '/',
     redirect: '/install'
   },
   {
     path: '/home',
     name: 'Home',
-    component: Home
-  }
+    component: Home,
+    beforeEnter: authGuard,
```
Comment (ymaheshwari1) [inline src/router/index.ts:45]: I think we don't need authGuard here.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/api/index.ts [hunk 11]
```diff
@@ -68,7 +68,8 @@ const api = async (customConfig: any) => {
         data: customConfig.data,
         params: customConfig.params
     }
-    const baseURL = process.env.VUE_APP_BASE_URL;
+    let baseURL = store.getters['user/getInstanceUrl'];
+        baseURL = baseURL && baseURL.startsWith('http') ? baseURL : `https://${baseURL}.hotwax.io/api/`;
```
Comment (adityasharma7) [inline src/api/index.ts]: Indentation issue

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**